contrib/mesos/pkg/executor/service/service.go

```diff
@@ -212,7 +215,90 @@ func (s *KubeletExecutorServer) Run(hks hyperkube.Interface, _ []string) error {
 		log.Warningf("Unknown Docker exec handler %q; defaulting to native", s.DockerExecHandlerName)
 		dockerExecHandler = &dockertools.NativeExecHandler{}
 	}
+	dockerClient := dockertools.ConnectToDockerOrDie(s.DockerEndpoint)
+
+	//TODO(jdef) either configure Watch here with something useful, or else
+	// get rid of it from executor.Config
+	kubeletFinished := make(chan struct{})
+	execUpdates := make(chan kubelet.PodUpdate, 1)
+	exec := executor.New(executor.Config{
+		Updates:         execUpdates,
+		APIClient:       apiclient,
+		Docker:          dockerClient,
+		SuicideTimeout:  s.SuicideTimeout,
+		KubeletFinished: kubeletFinished,
+		ShutdownAlert: func() {
+			if shutdownCloser != nil {
+				if e := shutdownCloser.Close(); e != nil {
+					log.Warningf("failed to signal shutdown to external watcher: %v", e)
+				}
+			}
+		},
+		ExitFunc: os.Exit,
+		PodStatusFunc: func(pod *api.Pod) (*api.PodStatus, error) {
+			if s.klet == nil {
+				return nil, fmt.Errorf("PodStatucFunc called before kubelet is initialized")
+			}
+
+			status, err := s.klet.GetRuntime().GetPodStatus(pod)
+			if err != nil {
+				return nil, err
+			}
+
+			status.Phase = kubelet.GetPhase(&pod.Spec, status.ContainerStatuses)
+			hostIP, err := s.klet.GetHostIP()
+			if err != nil {
+				log.Errorf("Cannot get host IP: %v", err)
+			} else {
+				status.HostIP = hostIP.String()
+			}
+			return status, nil
+		},
+		StaticPodsConfigPath: staticPodsConfigPath,
+	})
+
+	// initialize driver and initialize the executor with it
+	dconfig := bindings.DriverConfig{
+		Executor:         exec,
+		HostnameOverride: s.HostnameOverride,
```

Comment:
the node name situation is kind of a mess. will need to carefully review
the current state of the kubelet code.

On Mon, Aug 24, 2015 at 1:20 PM, Dr. Stefan Schimanski <
notifications@github.com> wrote:

> In contrib/mesos/pkg/executor/service/service.go
> <https://github.com/kubernetes/kubernetes/pull/13036#discussion_r37776231>
> :
>
> > +			status.Phase = kubelet.GetPhase(&pod.Spec, status.ContainerStatuses)
> > +			hostIP, err := s.klet.GetHostIP()
> > +			if err != nil {
> > +				log.Errorf("Cannot get host IP: %v", err)
> > +			} else {
> > +				status.HostIP = hostIP.String()
> > +			}
> > +			return status, nil
> > +		},
> > +		StaticPodsConfigPath: staticPodsConfigPath,
> > +	})
> > +
> > +	// initialize driver and initialize the executor with it
> > +	dconfig := bindings.DriverConfig{
> > +		Executor:         exec,
> > +		HostnameOverride: s.HostnameOverride,
>
> This is exactly the part where I need your critical eyes whether the
> restructuring broke anything ;-) I tried to avoid any behaviour change
> while refactoring that. But maybe I missed some implicit dependency.
>
> This work is also a preparation for a real split of the executor and the
> kubelet processes. So we have to decouple those initializations.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/kubernetes/kubernetes/pull/13036/files#r37776231>.
>